grails-app/domain/org/pih/warehouse/order/OrderAdjustment.groovy

```diff
@@ -111,13 +111,28 @@ class OrderAdjustment implements Serializable, Comparable<OrderAdjustment> {
         return invoiceItems?.findAll { it.invoice.isRegularInvoice && !it.inverse }?.sum { it.quantity } ?: 0
     }
 
+    /**
+     * Overall invoiced amount for this adjustment. Can be positive, negative or 0.
+     * */
+    BigDecimal getInvoicedAmount() {
+        return invoiceItems.findAll { it.invoice.isRegularInvoice && !it.inverse }?.sum { it.amount } ?: 0
```

Comment:
I think you can do it in one `sum`:
```
return invoiceItems.?.sum { it -> 
   if ( it.invoice.isRegularInvoice && !it.inverse) {
      return it.amount
   }

   return 0
} ?: 0
```

In the other functions, you can do the same, to avoid iterating twice over the collections.